File: Pyro.Nc/Simulation/CutResult.cs
using System;

namespace Pyro.Nc.Simulation
{
    public struct CutResult
    {
        public double TotalTime;
        public long TotalVerticesCut;
        public bool Threw;

        public CutResult(double totalTime, long totalVerticesCut)
        {
            TotalTime = totalTime;
            TotalVerticesCut = totalVerticesCut;
            Threw = false;
        }
        
        public CutResult(long totalTime, long totalVerticesCut, bool threw)
        {
            TotalTime = totalTime;
            TotalVerticesCut = totalVerticesCut;
            Threw = threw;
        }
    }
}

File: Pyro.Nc/RustSim3D/src/CutResult.rs
pub struct CutResult
{
    pub total_time: i64,
    pub total_vertices_cut: i64,
    pub threw: bool
}

impl CutResult{
    extern fn new(total_t: i64, total_verts_cut: i64, threw: bool) -> CutResult{
        return CutResult{
            total_time: total_t,
            total_vertices_cut: total_verts_cut,
            threw
        };
    }
}